src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -387,14 +387,21 @@ internal static bool IsMinimalProgressRenderingEnabled()
                 maxWidth = PSStyle.Instance.Progress.MaxWidth;
             }
 
+            // if the activity is really long, only use up to half the width
+            if (Activity.Length > maxWidth / 2)
+            {
+                Activity = Activity.Substring(0, maxWidth / 2) + PSObjectHelper.Ellipsis;
```

Comment:
My concern is that Activity is in ProgressRecord which could be reused over iterations. So we shouldn't change it.